lib/rest-api.php

```diff
@@ -316,7 +316,15 @@ function gutenberg_auto_draft_get_sample_permalink( $permalink, $id, $title, $na
  * @since 7.x.0
  */
 function gutenberg_register_image_editor() {
-	$image_editor = new WP_REST_Image_Editor_Controller();
-	$image_editor->register_routes();
+	global $wp_version;
+
+	// Strip -beta and -src from the version string. Messes up version_compare().
+	$version = explode( '-', $wp_version, 2 )[0];
+
+	// Only register routes for versions older than WP 5.5.
+	if ( version_compare( $version, '5.5', '<' ) ) {
+		$image_editor = new WP_REST_Image_Editor_Controller();
+		$image_editor->register_routes();
+	}
```

Comment:
[`version_compare` should handle prereleases well](https://www.php.net/manual/en/function.version-compare.php#refsect1-function.version-compare-notes), where we want to register for anything less than `5.5-beta`.

> Note:
> 
> Note that pre-release versions, such as 5.3.0-dev, are considered lower than their final release counterparts (like 5.3.0).

I'd expect this to work:

```suggestion
	// Only register routes for versions older than WP 5.5-beta.
	// This is handled by core endpoints starting with 5.5.
	if ( version_compare( $wp_version, '5.5-beta', '<' ) ) {
		$image_editor = new WP_REST_Image_Editor_Controller();
		$image_editor->register_routes();
	}
```

```php
var_export( version_compare( '5.5-beta', '5.5-beta', '<' ) );
// false
var_export( version_compare( '5.5-beta', '5.5', '<' ) );
// true
var_export( version_compare( '5.5-beta', '5.6', '<' ) );
// true
var_export( version_compare( '5.5-beta', '6.0-beta', '<' ) );
// true
var_export( version_compare( '5.5-beta', '6.0', '<' ) );
// true
var_export( version_compare( '5.5-beta', '5.4', '<' ) );
// false
```

Reply:
ah ok, I was reading the php docs and it wasn't quite clear that you could include `-beta`
I updated and it simplifies it nicely.

Reply:
I included removing the `-src` portion because that is done in the `gutenberg.php` script here https://github.com/WordPress/gutenberg/blob/master/gutenberg.php#L147